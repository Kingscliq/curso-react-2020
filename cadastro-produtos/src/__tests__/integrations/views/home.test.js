import "@testing-library/jest-dom";
import { render, act } from "./../../../../test-utils";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";

import Home from "./../../../views/home";
import React from "react";

let documentBody;

describe("Render correctly <Home />", () => {
  beforeEach(() => {
    documentBody = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );
  });

  it("should show the text welcome", () => {
    const { getByText } = documentBody;
    const text = getByText("Bem-vindo!");
    expect(text).toBeInTheDocument();
  });

  it("should show description system", () => {
    const { getByText } = documentBody;
    const description = getByText("Sistema de cadastro de produtos");
    expect(description).toBeInTheDocument();
  });

  it("should click in Cadastrar button", () => {
    const { getByRole } = documentBody;
    const btn = getByRole("button", {
      name: /Cadastrar/i,
    });
    act(() => {
      userEvent.click(btn);
    });
  });

  it("should be generate snapshot", () => {
    const { container } = documentBody;
    expect(container).toMatchSnapshot();
  });
});
