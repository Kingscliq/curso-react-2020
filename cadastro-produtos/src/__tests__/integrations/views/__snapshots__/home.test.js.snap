// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly <Home /> should be generate snapshot 1`] = `
<div>
  <div
    class="jumbotron"
  >
    <h1
      class="display-3"
    >
      Bem-vindo!
    </h1>
    <p
      class="lead"
    >
      Sistema de cadastro de produtos
    </p>
    <hr
      class="my-4"
    />
    <p
      class="lead"
    >
      <a
        class="btn btn-primary btn-lg"
        href="/cadastro-produto"
        role="button"
      >
        Cadastrar
      </a>
    </p>
  </div>
</div>
`;
